source/common/grpc/common.cc

```diff
@@ -195,21 +195,20 @@ Http::MessagePtr Common::prepareHeaders(const std::string& upstream_cluster,
                                         const std::string& method_name,
                                         const absl::optional<std::chrono::milliseconds>& timeout) {
   Http::MessagePtr message(new Http::RequestMessageImpl());
-  message->headers().insertMethod().value().setReference(Http::Headers::get().MethodValues.Post);
+  message->headers().setReferenceMethod(Http::Headers::get().MethodValues.Post);
   message->headers().insertPath().value().append("/", 1);
```

Comment:
So my thesis here is that you should remove the insert* methods as they return hard-to-reason-about non-const refs.

In this case IMO you could just have `setPath(absl::StrCat(...))` and we're good to go :)  IDK of something was previously in 'path' though.

And if there are other cases you should expose appendPath(absl::string_view)

Reply:
Nothing is previously in `:path` at this point, since `message` is created right there. 
Hmmm. So there are a few small tricky `insert` cases besides this one that I wanted to handle in a separate PR (`toGrpcTimeout` in particular). I'll handle the cases that can be moved to the functionality introduced in this PR though. I think that does make sense.

Reply:
A TODO to remove the insert() methods SGTM.

Reply:
Done.